src/main/app/controller/courts/PhotoController.ts

```diff
@@ -0,0 +1,130 @@
+import autobind from 'autobind-decorator';
+import {AuthedRequest} from '../../../types/AuthedRequest';
+import {Response} from 'express';
+import {Error} from '../../../types/Error';
+import {PhotoPageData} from '../../../types/PhotoPageData';
+import config from 'config';
+import {CSRF} from '../../../modules/csrf';
+import {AxiosError} from 'axios';
+import {CourtPhoto} from '../../../types/CourtPhoto';
+
+@autobind
+export class PhotoController {
+
+  getCourtPhotoErrorMsg = 'A problem occurred when retrieving the court photo. ';
+  putCourtPhotoErrorMsg = 'A problem occurred when updating the court photo. ';
+  deleteCourtPhotoErrorMsg = 'A problem occurred when deleting the court photo. ';
+  imageTypeError = 'File must be a JPEG or PNG.';
+  imageSizeError = 'File must be a less than 2mb.';
+
+  public async get(req: AuthedRequest, res: Response): Promise<void> {
+    await this.render(req, res);
+  }
+
+  public async put(req: AuthedRequest, res: Response): Promise<void> {
+    const imageFileName = req.body.name as string;
+    const slug: string = req.params.slug as string;
+    const fileType = req.body.fileType;
+    const oldCourtPhoto = req.body.oldCourtPhoto as string;
+    const imageFile = req.file;
+
+    if (fileType !== 'image/png' && fileType !== 'image/jpeg') {
+      return this.render(req, res, [this.imageTypeError], false, this.imageTypeError, null);
+    }
+
+    if (imageFile.size > 2000000) {
+      return this.render(req, res, [this.imageSizeError], false, this.imageSizeError, null);
+    }
+
+    if (!CSRF.verify(req.body.csrfToken)) {
+      return this.render(req, res, [this.putCourtPhotoErrorMsg], false);
+    }
+
+    await req.scope.cradle.azure.uploadImageFileToAzure(imageFile, imageFileName)
+      .then(async () => {
```

Comment:
can the parameters have type like imageFile: string , imageFileName : string

Reply:
This is where the method is being called but I have the parameter types where the method is declared in the AzureBlobStorage class.

public async uploadImageFileToAzure(file: File, imageFileName: string): Promise<void> {
    const blockBlobClient = this.containerClient.getBlockBlobClient(imageFileName);
...
}